src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1059,25 +1047,69 @@ private bool CheckPropertyValueKind(SyntaxNode node, BoundExpression expr, BindV
                 var baseReference = (BoundBaseReference)receiverOpt;
 
-                if (method.IsImplementableInterfaceMember() &&
-                    baseReference.ExplicitBaseReferenceOpt?.Type.IsInterfaceType() == true)
+                if (baseReference.ExplicitBaseReferenceOpt != null)
                 {
-                    if (TypeSymbol.TryFindBaseImplementationInInterface((NamedTypeSymbol)baseReference.ExplicitBaseReferenceOpt.Type, method, diagnostics, node, this) is null)
+                    if (baseReference.HasErrors)
+                    {
+                        return true;
+                    }
+
+                    TypeSymbol baseType = baseReference.ExplicitBaseReferenceOpt.Type;
+
+                    if (baseType.IsInterfaceType() && (member as MethodSymbol)?.IsImplementable() == true)
                     {
-                        Debug.Assert(!diagnostics.IsEmptyWithoutResolution);
+                        MultiDictionary<Symbol, Symbol>.ValueSet set = TypeSymbol.FindImplementationInInterface(member, (NamedTypeSymbol)baseType);
+
+                        if (set.Count == 1)
+                        {
+                            member = set.Single();
+
+                            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+                            if (!IsAccessible(member, ref useSiteDiagnostics, accessThroughType: null))
+                            {
+                                diagnostics.Add(node, useSiteDiagnostics);
+                                Error(diagnostics, ErrorCode.ERR_BadAccess, node, member);
+                                return true;
+                            }
+                            else if (member.IsAbstract)
+                            {
+                                Error(diagnostics, ErrorCode.ERR_AbstractBaseCall, node, member);
+                                return true;
+                            }
+                        }
+                        else if (!member.ContainingType.Equals(baseType, TypeCompareKind.AllIgnoreOptions))
+                        {
+                            Error(diagnostics, ErrorCode.ERR_NotImplementedInBase, node, member, baseType);
+                            return true;
+                        }
+                        else
+                        {
+                            Error(diagnostics, ErrorCode.ERR_AbstractBaseCall, node, member);
+                            return true;
+                        }
+
+                        return false;
+                    }
+                    else if (!member.ContainingType.Equals(baseType, TypeCompareKind.AllIgnoreOptions))
+                    {
+                        Error(diagnostics, ErrorCode.ERR_NotDeclaredInBase, node, member, baseType);
                         return true;
                     }
                 }
-                else if (!checkOnlyAccessThroughInterface && method.IsAbstract)
+
+                if (member.IsAbstract)
```

Comment:
>if (member.IsAbstract) [](start = 16, length = 22)

There was no impact of removing `checkOnlyAccessThroughInterface` (used to be set for assigning to base event)? Feels like we're producing an extra diagnostic.

Reply:
> There was no impact of removing checkOnlyAccessThroughInterface (used to be set for assigning to base event)? Feels like we're producing an extra diagnostic. 

That was compensated by the change in ```Binder.CheckEventValueKind```

---
In reply to: [266171739](https://github.com/dotnet/roslyn/pull/34121#discussion_r266171739) [](ancestors = 266171739)